### DISCUSSION TITLE
Differences in KinesisVideoClient between JS SDKs v2 and v3

### DISCUSSION THREAD

**Discussion by ModalityZ** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-js#333] [Q&A] [answered]
To future-proof my app, I would like to use [KinesisVideoClient in the v3 JS SDK](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/kinesis-video/) instead of [AWS.KinesisVideo in the v2 JS SDK](
https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-js?tab=readme-ov-file#class-sigv4requestsigner:~:text=new%20AWS.KinesisVideo). However, the v3 doc does not show how to provide credentials and the region. I tried
```
import * as mKvswClient_video from "@aws-sdk/client-kinesis-video";
...
mKvswClient_authenticated
      = new mKvswClient_video({ ...oAwsCredentialsAndRegion });
```
but that results in error `mKvswClient_video is not a constructor`

**Comment by sirknightj** [comment, marked as answer]
Please refer to the sample reactjs integration, which uses the AWS SDK JS v3: https://github.com/aws-samples/amazon-kinesis-video-streams-demos/blob/1b6c939b752615c934aa8b8f4ef1289af6fb17cd/webrtc-js/src/MasterView.tsx#L82

**Comment by ModalityZ** [comment]
It's odd that property `credentials` doesn't appear at https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/kinesis-video/#table-header-:r1k:
